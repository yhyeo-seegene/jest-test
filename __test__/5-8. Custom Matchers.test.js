describe('Custom Matchers', () => {
    const mockFunc = jest.fn();
    const arg1 = 1;
    const arg2 = 2;
  
    test('The mock function was called at least once', () => {
      mockFunc();
      expect(mockFunc).toHaveBeenCalled();
      expect(mockFunc.mock.calls.length).toBeGreaterThan(0);
    });
  
    test('The mock function was called at least once with the specified args', () => {
      mockFunc(arg1, arg2);
      expect(mockFunc).toHaveBeenCalledWith(arg1, arg2);
      // expect(mockFunc.mock.calls).toContainEqual([arg1, arg2]);
    });
  
    test('The last call to the mock function was called with the specified args', () => {
      mockFunc(3, 4);
      mockFunc(arg1, arg2);
      expect(mockFunc).toHaveBeenLastCalledWith(arg1, arg2);
      // expect(mockFunc.mock.calls[mockFunc.mock.calls.length - 1]).toEqual([
      //   arg1,
      //   arg2,
      // ]);
    });
  
    test('All calls and the name of the mock is written as a snapshot', () => {
      mockFunc(arg1, arg2);
      expect(mockFunc).toMatchSnapshot();
      // expect(mockFunc.mock.calls).toEqual([[arg1, arg2]]);
      // expect(mockFunc.getMockName()).toBe('a mock name');
    });
  });