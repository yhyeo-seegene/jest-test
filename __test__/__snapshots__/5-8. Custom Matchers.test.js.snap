// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Matchers All calls and the name of the mock is written as a snapshot 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
    Array [
      1,
      2,
    ],
    Array [
      3,
      4,
    ],
    Array [
      1,
      2,
    ],
    Array [
      1,
      2,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
